### PR TITLE
automator: Fix branch name guessing

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

**Comment by jwendell** [comment]
https://prow.maistra.io/view/gs/maistra-prow-testing/logs/envoy_2.0-update-proxy/1331297138696523776

File: tools/automator.sh [hunk 1]
```diff
@@ -96,7 +96,7 @@ function validate_args() {
   TOKEN="$(cat "${TOKEN_FILE}")"
   [ -z "${TOKEN}" ] && usage "Invalid token file"
 
-  BRANCH="${BRANCH:-$(git describe --contains --all HEAD)}"
+  BRANCH="${BRANCH:-$(git symbolic-ref --quiet --short HEAD)}"
```
Comment (dgn) [inline tools/automator.sh:99]: hmm.. when I run that on a repo that is in detached-head state I get empty string:

```bash
16:10:38 t580-rh {MAISTRA-1898} ~/dev/go/src/istio.io/istio$ git checkout 693f2fdcce7fcdc11925af2f0dd07259a57e899e
Note: switching to '693f2fdcce7fcdc11925af2f0dd07259a57e899e'.

You are in 'detached HEAD' state. [...]

HEAD is now at 693f2fdcce Fix merge logic and conflicts in push_context.go
16:10:41 t580-rh {(693f2fdcce...)} ~/dev/go/src/istio.io/istio$ git describe --contains --all HEAD
MAISTRA-1898~1
16:10:43 t580-rh {(693f2fdcce...)} ~/dev/go/src/istio.io/istio$ git symbolic-ref --quiet --short HEAD
16:10:46 t580-rh {(693f2fdcce...)} ~/dev/go/src/istio.io/istio$
```

Comment (jwendell) [reply tools/automator.sh:99]: yes. same here. This is not supposed to run in a detached head

Comment (jwendell) [reply tools/automator.sh:99]: btw, upstream ran into the same issue... https://github.com/istio/test-infra/pull/2611

Comment (dgn) [reply tools/automator.sh:99]: ah I see, thanks

**Review by dgn — APPROVED** [review]

**Pull request merged**